hawkbit-core/src/test/java/org/eclipse/hawkbit/exception/DelegatingConditionalErrorHandlerTest.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2015 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.exception;
+
+import io.qameta.allure.Description;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.util.ErrorHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Feature("Unit Tests - Delegating Conditional Error Handler")
+@Story("Delegating Conditional Error Handler")
+@RunWith(MockitoJUnitRunner.class)
+public class DelegatingConditionalErrorHandlerTest {
+
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
I think we better use the junit5 style here, since there are already efforts to migrate junit4 to junit5
```java
    @Test
    @Description("Verifies that with a list of conditional error handlers, the error is delegated to specific handler.")
    public void verifyDelegationHandling(){
        List<ConditionalErrorHandler> handlers = new ArrayList<>();
        handlers.add(new ConditionalErrorHandler1());
        handlers.add(new ConditionalErrorHandler2());
        Assertions.assertThrows(IllegalArgumentException.class,
        () -> new DelegatingConditionalErrorHandler(handlers, new DefaultErrorHandler()).handleError(new Throwable(new IllegalArgumentException())),
        "Expected handled exception to be of type IllegalArgumentException");
    }
```